specs/CustomStoragePartition.md

```diff
@@ -0,0 +1,208 @@
+Custom Storage Partition
+===
+
+# Background
+For certain WebView host apps, there is a desire to have different contexts for
+different WebViews using the same profile so that when the same websites run in
+these WebViews, they will have different DOM storages and cookie jars. An example
+is different accounts within an application.
+Previosly, the application can use different profiles for different contexts, but
+that has 2 short comings:
+- WebViews from different profiles are not allowed to have opener/opened window relationship.
+- Using profiles means totally different storage for all files like http caches and less performance.
+
+To help managing different application contexts while using the same profile, we are
+introducing an API to set custom storage parition and an API to clear all data in
+a custom storage partition.
+
+When an application sets custom storage parition for a WebView, the sites and iframes
+running inside the WebView will act as if the site were running in a third party iframe
+inside a top level site uniquelly associated with the custom storage partition id and have
+separate [storage partition](https://developer.chrome.com/docs/privacy-sandbox/storage-partitioning/)
+and [cookie partition](https://developer.chrome.com/docs/privacy-sandbox/chips/).
+
+# Conceptual pages (How To)
```

Comment:
Consider updating examples to do the following:

[1:26 PM] Raymond Chen
For example, if I'm a blog authoring tool, I might want to show two webviews, one in light mode and one in dark mode.

[1:26 PM] Raymond Chen
Or one as a signed-in user and one as an anonymous user.

[1:26 PM] Raymond Chen
(I would prefer that our examples not talk about different accounts.)

Reply:
Will use anonymous vs signed in user as example in background section. We have separate API for dark/light mode and dark mode is less linked to data in partition.